File: playwright-test/dashboard-check.js
const { chromium } = require('playwright');
const fs = require('fs');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  
  try {
    console.log('대시보드에 접속 중...');
    await page.goto('http://158.180.82.112:8080/', { waitUntil: 'networkidle' });
    
    // 페이지 로딩 대기
    await page.waitForTimeout(3000);
    
    console.log('초기 스크린샷 촬영 중...');
    await page.screenshot({ 
      path: 'dashboard-initial.png', 
      fullPage: true 
    });
    
    // 각 카드 확인
    const cards = [
      'System Status Card',
      'Account Information Card', 
      'Today\'s Performance Card',
      'Strategy Signal Strengths Card',
      'Recent Trades Card'
    ];
    
    for (const card of cards) {
      console.log(`${card} 확인 중...`);
      
      // 카드별 요소 확인
      try {
        if (card === 'System Status Card') {
          const statusElement = await page.$('.status-indicator');
          if (statusElement) {
            const status = await statusElement.textContent();
            console.log(`시스템 상태: ${status}`);
          }
        }
        
        if (card === 'Account Information Card') {
          const krwBalance = await page.$('[data-label="KRW Balance"]');
          const btcBalance = await page.$('[data-label="BTC Balance"]');
          if (krwBalance) {
            const krw = await krwBalance.textContent();
            console.log(`KRW 잔액: ${krw}`);
          }
          if (btcBalance) {
            const btc = await btcBalance.textContent();
            console.log(`BTC 잔액: ${btc}`);
          }
        }
        
        if (card === 'Today\'s Performance Card') {
          const dailyPL = await page.$('[data-label="Daily P&L"]');
          if (dailyPL) {
            const pl = await dailyPL.textContent();
            console.log(`일일 손익: ${pl}`);
          }
        }
        
        if (card === 'Strategy Signal Strengths Card') {
          const strategies = ['market-making', 'stat-arb', 'microstructure', 'momentum', 'mean-reversion'];
          for (const strategy of strategies) {
            const element = await page.$(`[data-strategy="${strategy}"]`);
            if (element) {
              const signal = await element.textContent();
              console.log(`${strategy} 신호: ${signal}`);
            }
          }
        }
        
      } catch (error) {
        console.log(`${card} 확인 중 오류: ${error.message}`);
      }
    }
    
    // 30초 동안 대시보드 모니터링
    console.log('30초간 업데이트 모니터링 시작...');
    for (let i = 0; i < 6; i++) {
      await page.waitForTimeout(5000);
      console.log(`${(i + 1) * 5}초 경과...`);
      
      // 중간 스크린샷
      if (i === 2) {
        await page.screenshot({ 
          path: 'dashboard-15sec.png', 
          fullPage: true 
        });
        console.log('15초 경과 스크린샷 촬영');
      }
    }
    
    console.log('최종 스크린샷 촬영 중...');
    await page.screenshot({ 
      path: 'dashboard-final.png', 
      fullPage: true 
    });
    
    console.log('대시보드 점검 완료');
    
  } catch (error) {
    console.error('대시보드 접속 실패:', error);
    await page.screenshot({ 
      path: 'dashboard-error.png', 
      fullPage: true 
    });
  }
  
  await browser.close();
})();

File: playwright-test/dashboard_check.py
#!/usr/bin/env python3

import asyncio
from playwright.async_api import async_playwright
import os

async def check_dashboard():
    async with async_playwright() as p:
        # Launch browser
        browser = await p.chromium.launch(headless=False)
        context = await browser.new_context(
            viewport={'width': 1920, 'height': 1080}
        )
        page = await context.new_page()
        
        try:
            print("Navigating to dashboard...")
            # Try different ports for the dashboard
            dashboard_urls = [
                'http://158.180.82.112:8080/',  # Default dashboard port
                'http://158.180.82.112:5000/',  # Web app port
                'http://158.180.82.112:8081/',  # Alternative port
            ]
            
            dashboard_url = None
            for url in dashboard_urls:
                try:
                    print(f"Trying to connect to {url}...")
                    await page.goto(url, timeout=10000)
                    dashboard_url = url
                    print(f"Successfully connected to {url}")
                    break
                except Exception as e:
                    print(f"Failed to connect to {url}: {e}")
                    continue
            
            if not dashboard_url:
                raise Exception("Could not connect to any dashboard port")
            
            # Wait for page to load
            await page.wait_for_load_state('networkidle', timeout=30000)
            
            # Take a full page screenshot first
            await page.screenshot(path='dashboard_full.png', full_page=True)
            print("Full dashboard screenshot saved as dashboard_full.png")
            
            # Try to find and focus on process monitor section
            process_monitor_selectors = [
                '[class*="process"]',
                '[id*="process"]', 
                'h2:has-text("Process")',
                'h3:has-text("Process")',
                '.monitor',
                '#monitor',
                '.status',
                '#status'
            ]
            
            process_section = None
            for selector in process_monitor_selectors:
                try:
                    element = await page.query_selector(selector)
                    if element:
                        print(f"Found process monitor section with selector: {selector}")
                        process_section = element
                        break
                except:
                    continue
            
            if process_section:
                # Scroll to the process monitor section
                await process_section.scroll_into_view_if_needed()
                await page.wait_for_timeout(2000)  # Wait for scroll
                
                # Take screenshot of the process monitor section
                await process_section.screenshot(path='process_monitor.png')
                print("Process monitor section screenshot saved as process_monitor.png")
            else:
                print("Could not locate specific process monitor section, using full page screenshot")
            
            # Get page title and basic info
            title = await page.title()
            print(f"Page title: {title}")
            
            # Check if page loaded successfully
            content = await page.content()
            if 'error' in content.lower() or 'not found' in content.lower():
                print("Warning: Page may contain error messages")
            else:
                print("Dashboard appears to be loading normally")
                
        except Exception as e:
            print(f"Error accessing dashboard: {e}")
            # Still try to take a screenshot of whatever loaded
            try:
                await page.screenshot(path='dashboard_error.png')
                print("Error screenshot saved as dashboard_error.png")
            except:
                pass
        
        finally:
            await browser.close()

if __name__ == "__main__":
    asyncio.run(check_dashboard())